openassessment/xblock/static/js/src/studio/oa_tiny_mce.js

```diff
@@ -23,30 +23,46 @@ export const oaTinyMCE = (options) => {
 
   const getFonts = () => CUSTOM_FONTS + STANDARD_FONTS;
   const baseAssetUrl = options.base_asset_url;
+  const staticUrl = '/static';
+
+  const sanitizeContent = (content) => {
+    // rewrite anchor link
+    let newContent = rewriteStaticLinks(
+      content,
+      `<a href="${staticUrl}`,
+      `<a href="${baseAssetUrl}`,
+    );
+
+    // rewrite image src
+    newContent = rewriteStaticLinks(
+      newContent,
+      `<img src="${staticUrl}`,
+      `<img src="${baseAssetUrl}`,
+    );
+
+    return newContent;
+  };
 
   const dataHandler = (key, ...urls) => (data) => {
-    if (data.src) {
-      /* jshint undef:false */
-      data.src = rewriteStaticLinks(data.src, ...urls);
+    if (data[key]) {
+      data[key] = rewriteStaticLinks(data[key], ...urls);
     }
   };
 
-  const setupTinyMCE = (ed) => {
-    ed.on('SaveImage', dataHandler('src', baseAssetUrl, '/static'));
-    ed.on('EditImage', dataHandler('src', '/static', baseAssetUrl));
-    ed.on('SaveLink', dataHandler('html', baseAssetUrl, '/static'));
-    ed.on('EditLink', dataHandler('html', '/static', baseAssetUrl));
```

Comment:
It looks like the original code wanted to translate from one format in one direction and to that same format in the other direction. Did we go about doing that incorrectly, before?

Said another way: how does this code behave in the following user story?
As a course author, I expect that when I write certain text into the platform and save it, then close the tab and return to edit it later, I find the same text I entered is still there.

Reply:
I was a bit confuse too. As far as I can tell, `SaveLink` and `EditLink` was added later. The `dataHandler` was written to work for image only. There is slight change I put in.
For how it ended up in reverse order for one of them, I am really not sure. I tested with `debugger` on, it doesn't seem to have a correct behavior.

Current behavior, if you input `/static/...` in the url, it will get transform immediately. I do not have opinion on when it should get transform. I was just trying to make work from the previous code.

Reply:
@leangseu-edx could you answer directly my question about the user story?

Reply:
When the user write text, it should return the same contents he entered.
If he input images or anchor, there will be modification if and only if the user are copying from `studio -> File & Upload -> Copy Urls for studio`. It has the specific format of `/static/....`.